java/core/src/java/org/apache/orc/impl/writer/WriterImplV2.java

```diff
@@ -373,7 +379,11 @@ private void flushStripe() throws IOException {
       OrcProto.StripeFooter.Builder builder =
           OrcProto.StripeFooter.newBuilder();
       if (writeTimeZone) {
-        builder.setWriterTimezone(TimeZone.getDefault().getID());
+        if (useUTCTimeZone) {
+          builder.setWriterTimezone(TimeZone.getTimeZone("UTC").getID());
```

Comment:
I'd be tempted to just use setWriterTimezone("UTC"), because we'll already fail if UTC is called something else.